### PR TITLE
Add default strategy for problem

### PR DESCRIPTION
**Pull request by MTsatsarov**
Closes https://github.com/SoftUni-Internal/exam-systems-issues/issues/928

Added new column in Problem table.
On update and create Views for problem, new dropdown list is added which will hold the values of the selected submission types.
The Selected index of the strategy represents the default submission type for the problem.
On code editor page , in the dropdown list for submission types. the default for the problem will be selected automatically (TimeLimit, MemoryLimit.. etc will be updated).

### CODE REVIEW

File: Open Judge System/Data/OJS.Data.Models/Problem.cs [hunk 1]
```diff
@@ -62,6 +62,8 @@ public class Problem : DeletableEntity, IOrderable
         [DefaultValue(false)]
         public bool ShowDetailedFeedback { get; set; }
 
+        public string DefaultSubmissionTypeName { get; set; }
```
Comment (gogo4ds) [inline Open Judge System/Data/OJS.Data.Models/Problem.cs:65]: This better be Id. What will happen if the submission type is renamed?

**Review by gogo4ds — CHANGES_REQUESTED** [review]

File: Open Judge System/Data/OJS.Data.Models/Problem.cs (hunk 1 above)
Comment (MTsatsarov) [reply Open Judge System/Data/OJS.Data.Models/Problem.cs]: Done

File: Open Judge System/Data/OJS.Data.Models/Problem.cs [hunk 2]
```diff
@@ -62,6 +62,8 @@ public class Problem : DeletableEntity, IOrderable
         [DefaultValue(false)]
         public bool ShowDetailedFeedback { get; set; }
 
+        public int DefaultSubmissionTypeId { get; set; }
```
Comment (gogo4ds) [inline Open Judge System/Data/OJS.Data.Models/Problem.cs:65]: Why not nullable id? It is not required to have a default submission type. Also you can add navigational property for ease of use.

**Review by gogo4ds — CHANGES_REQUESTED** [review]

File: Open Judge System/Data/OJS.Data.Models/Problem.cs (hunk 2 above)
Comment (MTsatsarov) [reply Open Judge System/Data/OJS.Data.Models/Problem.cs:65]: Do we need navigational property for the defaultSubmissionType. We already have many to many relation and can extract it from there.

Comment (gogo4ds) [reply Open Judge System/Data/OJS.Data.Models/Problem.cs:65]: The relation is already made, so the navigational property will be just for convenience and it's not bad to have it.

Comment (MTsatsarov) [reply Open Judge System/Data/OJS.Data.Models/Problem.cs:65]: Done

File: Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/ProblemsController.cs [hunk 3]
```diff
@@ -886,12 +888,12 @@ private ViewModelType PrepareProblemViewModelForEdit(int id)
...
 
             var contest = problemEntity.FirstOrDefault()?.ProblemGroup.Contest;
-
             if (problem == null || contest == null)
             {
                 return null;
             }
-
+            
+            problem.DefaultSubmissionTypeId = problemEntity.FirstOrDefault().DefaultSubmissionTypeId;
```
Comment (gogo4ds) [inline Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/ProblemsController.cs:896]: A couple lines above this I see the same  `problemEntity.FirstOrDefault()` call.

**Review by gogo4ds — APPROVED** [review]

File: Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/ProblemsController.cs (hunk 3 above)
Comment (MTsatsarov) [reply Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/ProblemsController.cs]: fixed

**Review by gogo4ds — APPROVED** [review]

**Pull request merged**